homeassistant/components/ntfy/notify.py

```diff
@@ -0,0 +1,153 @@
+
+    if username and password:
+        auth = requests.auth.HTTPBasicAuth(username, password)
+    else:
+        auth = None
+
+    return NtfyNotificationService(
+        hass,
+        url,
+        topic,
+        auth,
+        verify_ssl,
+    )
+
+
+class NtfyNotificationService(BaseNotificationService):
+    """Implementation of a notification service for ntfy."""
+
+    def __init__(
+        self,
+        hass,
+        url,
+        topic,
+        auth,
+        verify_ssl,
+    ):
+        """Initialize the service."""
+        self._hass = hass
+        self._url = url
+        self._topic = topic
+        self._auth = auth
+        self._verify_ssl = verify_ssl
+
+    def send_message(self, message: str = "", **kwargs: Any) -> None:
+        """Send a message to a user."""
+
+        post_data = {
+            "topic": self._topic,
+            "message": message,
+            "title": kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT),
+        }
+
+        # Ignore target
+
+        extra_data = kwargs.get(ATTR_DATA, {})
+        # If additional data parameters provided by user
+        if extra_data:
+            # Add message and topic to kwargs
+            kwargs[ATTR_MESSAGE] = message
+            kwargs[CONF_TOPIC] = self._topic
+
+            # Title already exists in kwargs
+
+            def _data_template_creator(value):
+                """Recursive template creator helper function."""
+                if isinstance(value, dict):
+                    return value
+
+                # Parsing error (data is not a dictionary)
+                value.hass = self._hass
+                return value.async_render(kwargs, parse_result=False)
+
+            # Combine additional data with message and title in JSON body
+            post_data.update(_data_template_creator(extra_data))
+
+        response = requests.post(
+            self._url,
+            json=post_data,
+            timeout=10,
+            auth=self._auth,
+            verify=self._verify_ssl,
+        )
+
+        if (
+            response.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR
+            and response.status_code < 600
+        ):
+            _LOGGER.exception(
+                "Server error. Response %d: %s:", response.status_code, response.reason
+            )
+        elif (
+            response.status_code >= HTTPStatus.BAD_REQUEST
+            and response.status_code < HTTPStatus.INTERNAL_SERVER_ERROR
+        ):
+            _LOGGER.exception(
+                "Client error. Response %d: %s:", response.status_code, response.reason
+            )
+        elif (
+            response.status_code >= HTTPStatus.OK
+            and response.status_code < HTTPStatus.MULTIPLE_CHOICES
+        ):
+            _LOGGER.debug(
+                "Success. Response %d: %s:", response.status_code, response.reason
+            )
+        else:
+            _LOGGER.debug("Response %d: %s:", response.status_code, response.reason)
```

Comment:
I'd probably just go with

```suggestion
        try:
            resp.raise_for_status()
        except requests.HTTPError as he:
            _LOGGER.exception(str(he))
        else:
            _LOGGER.debug(
                "Success. Response %d: %s", response.status_code, response.reason
            )
```
and logging what i assume is status 200 would also maybe be unnecessary...

Reply:
Good idea. Was largely copy-pasting from the REST notify integration. Going to utilize requests truthiness.